src/IO/ParallelReadBuffer.cpp

```diff
@@ -265,8 +261,12 @@ void ParallelReadBuffer::finishAndWait()
 {
     emergency_stop = true;
 
-    std::unique_lock lock{mutex};
-    readers_done.wait(lock, [&] { return active_working_reader == 0; });
+    size_t active_readers = active_working_reader.load();
+    while (active_readers != 0)
+    {
+        active_working_reader.wait(active_readers);
```

Comment:
It's good to know that atomics have `wait` function. 👌 
Is there in correctness reason for this change?

Reply:
Correctness was ok.